system/include/emscripten/bind.h

```diff
@@ -1374,13 +1408,11 @@ namespace emscripten {
         };
     }
 
-    template<typename T>
-    class_<std::vector<T>> register_vector(const char* name) {
-        typedef std::vector<T> VecType;
-
+    template<typename T,typename VecType=std::vector<T>>
```

Comment:
Why did you generalize this?  It may preclude us from binding other methods of std::vector without breaking backwards compatibility.  (Someone may bind a different vector type that does not bind all the methods of std::vector.)